fiftyone/utils/activitynet.py

```diff
@@ -0,0 +1,1038 @@
+        a100_data_map = {
+            to_uuid(p): os.path.join(a100_data_path, p)
+            for p in etau.list_files(a100_data_path, recursive=True)
+            if not (p.endswith(".part") or p.endswith(".ytdl"))
+        }
+        etas.write_json(
+            a100_data_map,
+            self.a100_info.data_json_path(split),
+            pretty_print=True,
+        )
+
+        a200_data_path = self.a200_info.data_dir(split)
+        a200_data_map = {
+            to_uuid(p): os.path.join(a200_data_path, p)
+            for p in etau.list_files(a200_data_path, recursive=True)
+            if not (p.endswith(".part") or p.endswith(".ytdl"))
+        }
+        a200_data_map.update(a100_data_map)
+        etas.write_json(
+            a200_data_map,
+            self.a200_info.data_json_path(split),
+            pretty_print=True,
+        )
+
+    @classmethod
+    def from_dataset_dir(cls, dataset_dir, version):
+        dataset_dir = os.path.abspath(dataset_dir)
+        foz_dir, _, _ = ActivityNetDatasetInfo.get_dir_info(dataset_dir)
+        return cls(foz_dir, version)
+
+
+class ActivityNetDatasetInfo(object):
+    """Class that stores information related to paths, labels, and sample IDs
+    for an ActivityNet dataset download.
+    """
+
+    def __init__(self, foz_dir):
+        self.foz_dir = os.path.abspath(foz_dir)
+
+        self.raw_annotations = self._get_raw_annotations()
+        self.taxonomy = self.raw_annotations["taxonomy"]
+        self.all_classes = self.get_all_classes(self.taxonomy)
+        self.ensure_data_dirs()
+
+        self._splitwise_sample_ids = self._parse_sample_ids()
+        self.all_sample_ids = _flatten_list(
+            self._splitwise_sample_ids.values()
+        )
+
+        self.update_existing_sample_ids()
+
+    @property
+    def splits(self):
+        return ["train", "test", "validation"]
+
+    @property
+    def version(self):
+        raise NotImplementedError("Subclass must implement version")
+
+    @property
+    def dataset_dir(self):
+        return os.path.join(self.foz_dir, "activitynet-%s" % self.version)
+
+    @property
+    def raw_anno_path(self):
+        return os.path.join(self.dataset_dir, "raw_labels.json")
+
+    def split_sample_ids(self, split):
+        return self._splitwise_sample_ids[split]
+
+    def existing_split_sample_ids(self, split):
+        return self._splitwise_existing_sample_ids[split]
+
+    def split_dir(self, split):
+        return os.path.join(self.dataset_dir, split)
+
+    def data_dir(self, split):
+        return os.path.join(self.split_dir(split), "data")
+
+    def data_json_path(self, split):
+        return os.path.join(self.split_dir(split), "data.json")
+
+    def labels_path(self, split):
+        return os.path.join(self.split_dir(split), "labels.json")
+
+    def error_path(self, split):
+        return os.path.join(self.split_dir(split), "download_errors.json")
+
+    def ensure_data_dirs(self):
```

Comment:
nit: I noticed that the split directories are created regardless of whether I've actually downloaded any content for that split. All other zoo datasets only create the directories if the split has actually been downloaded

I sometimes check the directory structure as a quick way of seeing what splits I've downloaded. Would be nice if this were possible here, too